step/step_test.go

```diff
@@ -0,0 +1,96 @@
+package step
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"testing"
+
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+)
+
+func TestRemoteCacheStep_Run(t *testing.T) {
+	tests := []struct {
+		name    string
+		envRepo env.Repository
+		wantErr bool
+	}{
+		{
+			name: "happy path",
+			envRepo: fakeEnvRepo{envVars: map[string]string{
+				"BITRISEIO_BITRISE_SERVICES_ACCESS_TOKEN": "fake access token",
+				"verbose": "false",
+			}},
+		},
+		{
+			name: "missing auth token",
+			envRepo: fakeEnvRepo{envVars: map[string]string{
+				"verbose": "false",
+			}},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fakePath := t.TempDir()
+			step := RemoteCacheStep{
+				logger:         log.NewLogger(),
+				inputParser:    stepconf.NewInputParser(tt.envRepo),
+				commandFactory: command.NewFactory(tt.envRepo),
+				envRepo:        tt.envRepo,
+				pathModifier:   fakePathModifier{path: fakePath},
+			}
+			err := step.Run()
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Run() error = %v, wantErr %v", err, tt.wantErr)
+			} else if err == nil {
+				_, err := ioutil.ReadFile(path.Join(fakePath, "init.gradle"))
```

Comment:
GoLand is telling me that the `ReadFile` method is deprecated. It links to [this documentation page](https://pkg.go.dev/io/ioutil#ReadFile), which says:

> Deprecated: As of Go 1.16, this function simply calls os.ReadFile.

Think we should switch it?

```suggestion
				_, err := os.ReadFile(path.Join(fakePath, "init.gradle"))
```

Reply:
Huh, I was wondering why I didn't get this suggestion from GoLand. Then I figured it out: I'm using Go 1.17.11 (the lowest version still installed on our stacks), where it's just a note without the `Deprecated:` prefix :)

Reply:
Anyway, I replaced the function call, take another look!

Reply:
Ah yep, I'm running Go 19.0.1 currently.